pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -191,13 +191,18 @@ func preempt(ctx context.Context, fh framework.FrameworkHandle, state *framework
 // considered for preemption.
 // We look at the node that is nominated for this pod and as long as there are
 // terminating pods on the node, we don't consider this for preempting more pods.
-func podEligibleToPreemptOthers(pod *v1.Pod, nodeInfos framework.NodeInfoLister) bool {
+func podEligibleToPreemptOthers(pod *v1.Pod, nodeInfos framework.NodeInfoLister, m framework.NodeToStatusMap) bool {
 	if pod.Spec.PreemptionPolicy != nil && *pod.Spec.PreemptionPolicy == v1.PreemptNever {
 		klog.V(5).Infof("Pod %v/%v is not eligible for preemption because it has a preemptionPolicy of %v", pod.Namespace, pod.Name, v1.PreemptNever)
 		return false
 	}
 	nomNodeName := pod.Status.NominatedNodeName
 	if len(nomNodeName) > 0 {
+		if status, ok := m[nomNodeName]; ok {
+			if status.Code() == framework.UnschedulableAndUnresolvable {
+				return true
+			}
+		}
```

Comment:
I don't think this logic can help resolve the problem thoroughly. There are more cases that this fix doesn't cover:

- The nominatedNode can no longer exist, due to network partition or node shutdown.
- High-priority pod gets scheduled to that node to take the "reserved" room, etc. And also the status.Code()
- Terminating low-priority pods are still occupying the "to-be-releasing" resources.

With that said, ok can be either true or false, also the code can be Unschedulable or UnschedulableAndUnresolvable. So fixing inside `podEligibleToPreemptOthers()` is not a good idea.

A promising solution is to continue the work in #91750 - which was too aggressive, but should be easy to rework a bit to thoroughly considering all meaningful return paths of preemption.

Reply:
> Just mention that the issue shouldn't only about the case of the node being down, it covers all the case of UnschedulableAndUnresolvable. Like the operator taint the node in same time, the pod won't be able to this node anymore, so the pod should be trying to preempt other node.

Ah, I understood your motivation now. Got misled by the issue that you highlighted as "the node gets down"... The solution here is totally irrelevant with the node is down or not, right?

I _guess_ (please confirm) you were trying to say: if the nominated node is somehow infeasible, in particular UnschedulableAndUnresolvable (it can be a new high-priority pod makes the preemptor violate some critical constraints , or the node gets tainted, etc.), we shouldn't return false in `podEligibleToPreemptOthers()` by simply checking whether there are pods being terminated, as that would waste one preemption cycle.

Reply:
> > Just mention that the issue shouldn't only about the case of the node being down, it covers all the case of UnschedulableAndUnresolvable. Like the operator taint the node in same time, the pod won't be able to this node anymore, so the pod should be trying to preempt other node.
> 
> Ah, I understood your motivation now. Got misled by the issue that you highlighted as "the node gets down"... The solution here is totally irrelevant with the node is down or not, right?

Yes, it isn't only about the node down, it should be about the nominated node considered as 'UnschedulableAndUnresolvable'. Sorry for miss leading that.

> 
> I _guess_ (please confirm) you were trying to say: if the nominated node is somehow infeasible, in particular UnschedulableAndUnresolvable (it can be a new high-priority pod makes the preemptor violate some critical constraints , or the node gets tainted, etc.), we shouldn't return false in `podEligibleToPreemptOthers()` by simply checking whether there are pods being terminated, as that would waste one preemption cycle.

Yes, exactly

Reply:
Got it. Then I think this PR is valid, could you reword the original issue and this PR's description?

Reply:
> Got it. Then I think this PR is valid, could you reword the original issue and this PR's description?

Thanks, done